src/Raven.Server/Smuggler/Documents/SmugglerExporter.cs

```diff
@@ -50,20 +50,47 @@ public ExportResult Export(DocumentsOperationContext context, Stream destination
                 writer.WritePropertyName(("BuildVersion"));
                 writer.WriteInteger(40000);
 
-                if (OperateOnTypes.HasFlag(DatabaseItemType.Documents))
+                if (Options.OperateOnTypes.HasFlag(DatabaseItemType.Documents))
                 {
+                    progress.Progress = "Exporting Documents";
+                    onProgress?.Invoke(progress);
                     writer.WriteComma();
                     writer.WritePropertyName(("Docs"));
-                    var documents = DocumentsLimit.HasValue
-                        ? _database.DocumentsStorage.GetDocumentsFrom(context, StartDocsEtag ?? 0, 0, DocumentsLimit.Value)
-                        : _database.DocumentsStorage.GetDocumentsFrom(context, StartDocsEtag ?? 0);
+
+                    IEnumerable<Document> documents = Options.CollectionsToExport.Count != 0 ? 
+                        _database.DocumentsStorage.GetDocumentsFrom(context, Options.CollectionsToExport, StartDocsEtag ?? 0, Options.BatchSize) : 
+                        _database.DocumentsStorage.GetDocumentsFrom(context, StartDocsEtag ?? 0, 0, Options.BatchSize);
```

Comment:
Do we still need BatchSize? What will happen if we will have more then BatchSize (1024) docs?